.circleci/config.yml

```diff
@@ -0,0 +1,240 @@
+    save_cache:
+      name: Save Upstream Cache
+      key: upstream-cache-v1-{{ arch }}-{{ checksum "/opt/ros_ws/checksum.txt" }}
+      paths:
+        - /opt/ros_ws
+  install_dependencies: &install_dependencies
+    run:
+        name: Install Dependencies
+        command: |
+          cat $ROS_WS/checksum.txt > $MOVEIT2_WS/checksum.txt
+          find $MOVEIT2_WS/src -not -iwholename '*.git/*' -type f -exec sha256sum {} \; | \
+            sort -k 2 | \
+            cut -d" " -f1 | \
+            sha256sum | \
+            awk '{print "source "$1}' >> $MOVEIT2_WS/checksum.txt
+          apt-get update
+          dependencies=$(
+            rosdep install -q -y \
+              --from-paths \
+                $ROS_WS/src \
+                src \
+              --ignore-src \
+              --verbose | \
+            awk '$1 ~ /^resolution\:/' | \
+            awk -F'[][]' '{print $2}' | \
+            tr -d \, | xargs -n1 | sort -u | xargs)
+          dpkg -s $dependencies | \
+            sha256sum | \
+            awk '{print "dependencies "$1}' >> $MOVEIT2_WS/checksum.txt
+  store_build_checksum: &store_build_checksum
+    store_artifacts:
+        path: checksum.txt
+  restore_build_cache: &restore_build_cache
+    restore_cache:
+      name: Restore Build Cache
+      key: build-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "/opt/moveit2_ws/checksum.txt" }}
+  source_build: &source_build
+    run:
+      name: Source Build
+      command: |
+        if [ -d install ]
+        then
+            echo "Skipping Source Build"
+        else
+            . $ROS_WS/install/setup.sh
+            colcon build \
+              --symlink-install \
+              --cmake-args \
+                -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
+                -DCOVERAGE_ENABLED=$COVERAGE_ENABLED
+        fi
+  save_build_cache: &save_build_cache
+    save_cache:
+      name: Save Build Cache
+      key: build-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "/opt/moveit2_ws/checksum.txt" }}
+      paths:
+        - /opt/moveit2_ws
+  copy_build_logs: &copy_build_logs
+    run:
+        name: Copy Build Logs
+        when: always
+        command: cp -rH log/latest_build log/build
+  store_build_logs: &store_build_logs
+    store_artifacts:
+        path: log/build
+  test_build: &test_build
+    run:
+        name: Test Build
+        command: |
+          . install/setup.sh
+          src/moveit2/tools/run_test_suite.bash
+  copy_test_logs: &copy_test_logs
+    run:
+        name: Copy Test Logs
+        when: always
+        command: cp -rH log/latest_test log/test
+  store_test_logs: &store_test_logs
+    store_artifacts:
+        path: log/test
+  report_code_coverage: &report_code_coverage
+    run:
+        name: Report Code Coverage
+        command: |
+          if [ "$COVERAGE_ENABLED" = "True" ]
+          then
+            . install/setup.sh
+            src/moveit2/tools/code_coverage_report.bash codecovio
```

Comment:
rather than adding another folder (``tools``) let's put this in our scripts folder:
https://github.com/ros-planning/moveit2/tree/master/moveit/scripts